src/psyclone/tests/domain/gocean/transformations/kern_trans_test.py

```diff
@@ -0,0 +1,309 @@
+    supplied node is not a Container node.
+
+    '''
+    kernel_psyir = fortran_reader.psyir_from_source(PROGRAM)
+    kern_trans = KernTrans()
+    kern_trans.metadata_name = "compute_cu"
+    routine = kernel_psyir.children[0].children[0]
+    routine.detach()
+    with pytest.raises(TransformationError) as info:
+        kern_trans.validate(routine)
+    assert ("Error in KernTrans transformation. The supplied call argument "
+            "should be a Container node but found 'Routine'."
+            in str(info.value))
+
+
+def test_validate_parent(fortran_reader):
+    '''Test that the validate method raises the expected exception if the
+    supplied node has a parent.
+
+    '''
+    kernel_psyir = fortran_reader.psyir_from_source(PROGRAM)
+    _ = FileContainer.create("test", SymbolTable(), [kernel_psyir])
+    kern_trans = KernTrans()
+    kern_trans.metadata_name = "compute_cu"
+    with pytest.raises(TransformationError) as info:
+        kern_trans.validate(kernel_psyir)
+    assert ("Error in KernTrans transformation. The supplied node should be "
+            "the root of a PSyIR tree but this node has a parent "
+            "(FileContainer)." in str(info.value))
+
+
+def test_validate_ok(fortran_reader):
+    '''Test that the validate method accepts valid psyir.'''
+    kernel_psyir = fortran_reader.psyir_from_source(PROGRAM)
+    kern_trans = KernTrans()
+    kern_trans.metadata_name = "compute_cu"
+    kern_trans.validate(kernel_psyir)
+
+
+def test_metadataname():
+    '''Test that the metadata_name getter and setter methods work as
+    expected.
+
+    '''
+    kern_trans = KernTrans()
+    assert kern_trans.metadata_name is None
+    with pytest.raises(TypeError) as info:
+        kern_trans.metadata_name = 0
+    assert ("The kern_trans transformation requires the metadata name to be "
+            "a string, but found 'int'." in str(info.value))
+    kern_trans.metadata_name = "hello"
+    assert kern_trans.metadata_name == "hello"
+
+
+def test_apply_validate(fortran_reader, monkeypatch):
+    '''Test that the validate method is called from the apply method. We
+    can't make make the metadata invalid as this would be picked up
+    when we specialise the symbol, even if validate were not
+    called. Therefore we monkeypatch the validate method.
+
+    '''
+    kernel_psyir = fortran_reader.psyir_from_source(PROGRAM)
+    kern_trans = KernTrans()
+
+    def dummy(_, options=None):
+        '''Dummy method used to check that the validate method is called from
+        the apply method.
+
+        '''
+        raise RuntimeError("dummy called")
+
+    monkeypatch.setattr(kern_trans, "validate", dummy)
+    kern_trans.metadata_name = "compute_cu"
+    with pytest.raises(RuntimeError) as info:
+        kern_trans.apply(kernel_psyir)
+    assert "dummy called" in str(info.value)
+
+
+def test_apply_keyerror(fortran_reader):
+    '''Test that the PSyIR tree walk to find the metadata symbol works if
```

Comment:
We should remove the duplicated symbol search functionality (by putting it into a utility method perhaps) so that this test isn't needed (as it's pretty much the same as for `validate`).

Reply:
I've modified the searches as you requested in earlier comments. Do you still want me to put the search into a utility method?

Reply:
Yes please.

Reply:
OK.